frontend/public/components/namespace.jsx

```diff
@@ -1048,106 +1114,109 @@ const namespaceBarDropdownDispatchToProps = (dispatch) => ({
       showStartGuide(noProjects);
     }
+  }, [namespace.data, namespace.loaded, showStartGuide]);
+
+  if (flagPending(canListNS)) {
+    return null;
   }
 
-  render() {
-    const {
-      activeNamespace,
-      onNamespaceChange,
-      setActiveNamespace,
-      canListNS,
-      canCreateProject,
-      useProjects,
-      children,
-      disabled,
-      t,
-    } = this.props;
-    if (flagPending(canListNS)) {
-      return null;
-    }
+  const { loaded, data } = namespace;
+  const model = getModel(useProjects);
+  const allNamespacesTitle =
+    model.label === 'Project' ? t('public~All Projects') : t('public~All Namespaces');
 
-    const { loaded, data } = this.props.namespace;
-    const model = getModel(useProjects);
-    const allNamespacesTitle =
-      model.label === 'Project' ? t('public~All Projects') : t('public~All Namespaces');
-    const items = {};
-    if (canListNS) {
-      items[ALL_NAMESPACES_KEY] = allNamespacesTitle;
-    }
+  let title = activeNamespace;
+  if (activeNamespace === ALL_NAMESPACES_KEY) {
+    title = allNamespacesTitle;
+  }
 
-    _.map(data, 'metadata.name')
-      .sort()
-      .forEach((name) => (items[name] = name));
+  const defaultActionItem = canCreateProject
+    ? [
+        {
+          actionTitle:
+            model.label === 'Project' ? t('public~Create Project') : t('public~Create Namespace'),
+          actionKey: CREATE_NEW_RESOURCE,
+        },
+      ]
+    : [];
+
+  const items = data.map((item) => ({ title: item.metadata.name, key: item.metadata.name }));
+
+  if (loaded && !items.find((option) => option.title === title) && title !== allNamespacesTitle) {
+    // Add current namespace if it isn't included
+    items.push({ title, key: title });
+  }
 
-    let title = activeNamespace;
-    if (activeNamespace === ALL_NAMESPACES_KEY) {
-      title = allNamespacesTitle;
-    } else if (loaded && !_.has(items, title)) {
-      // If the currently active namespace is not found in the list of all namespaces, put it in anyway
-      items[title] = title;
+  items.sort((a, b) => {
+    if (a.title === b.title) {
+      return 0;
     }
-    const defaultActionItem = canCreateProject
-      ? [
-          {
-            actionTitle:
-              model.label === 'Project' ? t('public~Create Project') : t('public~Create Namespace'),
-            actionKey: CREATE_NEW_RESOURCE,
-          },
-        ]
-      : [];
-
-    const onChange = (newNamespace) => {
-      if (newNamespace === CREATE_NEW_RESOURCE) {
-        createProjectModal({
-          blocking: true,
-          onSubmit: (newProject) => {
-            setActiveNamespace(newProject.metadata.name);
-            removeQueryArgument('project-name');
-          },
-        });
-      } else {
-        onNamespaceChange && onNamespaceChange(newNamespace);
-        setActiveNamespace(newNamespace);
-        removeQueryArgument('project-name');
-      }
-    };
+    return a.title > b.title ? 1 : -1;
```

Comment:
You should be able to use `String.localeCompare` if `title` is never undefined

```ts
items.sort((a, b) => a.title.localeCompare(b.title));
```

Reply:
Changed in next commit.